### PR TITLE
IPP-419: Updated logic app to output the id. Added 2 new modules: key_vault_secret_data to retrieve the secret value from a key vault, and logic_app_trigger_http_request_data to retrieve the callback url from a logic app's http trigger.

### PR DESCRIPTION
**Pull request by gerkestam2**
Kleine terugdonatie vanuit het DFN project.

### CODE REVIEW

**Review by wvanasten — APPROVED** [review]

**Review by jvanenckevort — CHANGES_REQUESTED** [review]
key_vault_secret module does the same thing, so key_vault_secret_data module is duplicate. If you need a different mechanism (e.g. multiple secrets or key_vault_name support), please implement it in that module. And you can keep me as a codeowner, that is why I put my name in there. I am still working on this repository as external collaborator :)

**Comment by gerkestam2** [comment]
Feedback verwerkt

**Review by jvanenckevort — CHANGES_REQUESTED** [review]
Small semantic comment, could you enable the Terraform formatter and check the format on save option? This way the code will be auto formatted and better readable.

**Comment by gerkestam2** [comment]
Which terraform formatter? I am using IntelliJ, with the following (default) formatting rules:
Screenshot 2022-11-14 at 11 26 31

File: modules/azure/logic_app_trigger_http_request_data/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+provider "azapi" {
+}
+
+provider "azurerm" {
+  features {}
+}
+
+// See also: https://github.com/hashicorp/terraform-provider-azurerm/issues/18866
+data "azapi_resource_action" "logicapp_callbackurl" {
```
Comment (jvanenckevort) [inline modules/azure/logic_app_trigger_http_request_data/main.tf:23]: These "=" should be aligned when you have the correct formatter installed and format on save is on. I am not sure what the exact settings are in intellij.

**Review by jvanenckevort — CHANGES_REQUESTED** [review]

File: modules/azure/logic_app_trigger_http_request_data/main.tf (hunk 1 above)
Comment (gerkestam2) [reply modules/azure/logic_app_trigger_http_request_data/main.tf:23]: thx, reformatted the code in this file, and enabled autoformatting on save.

**Comment by gerkestam2** [comment]
@Justin-Booij : kan deze geapproved / gemerged worden? Feedback van Jules is verwerkt.

**Comment by bartwesselink** [comment]
@Justin-Booij @gerkestam2 ik ga deze force mergen, comment van @jvanenckevort is opgelost.

**Comment by jvanenckevort** [comment]
> @Justin-Booij @gerkestam2 ik ga deze force mergen, comment van @jvanenckevort is opgelost.

Just wanted to approve :)

**Pull request merged**